Created nested utils.cpp

drisspg opened the pull request:
# Summary
This refactors the preprocessing for nestedtensors that glue into SDPA. This is done in order to aid with reviewing:
https://github.com/pytorch/pytorch/pull/97485

cc @cpuhrsch @jbschlosser @bhosmer

drisspg commented:
@pytorchbot merge

drisspg commented:
@pytorchbot merge -f "unrelated failures"

clee2000 commented:
@pytorchbot revert -m "sorry but this is breaking internal builds due to the new header file D49346814" -c ghfirst

drisspg committed:
Created nested utils.cpp (#109304)

Summary:
This refactors the preprocessing for nestedtensors that glue into SDPA. This is done in order to aid with reviewing:
https://github.com/pytorch/pytorch/pull/97485

cc cpuhrsch jbschlosser bhosmer


Differential Revision: D49374910

Pulled By: drisspg
